### DISCUSSION TITLE
Is single socket CPU count capped at 64?

### DISCUSSION THREAD

**Discussion by felgryn** [microsoft/WSL#11028] [Q&A]
Hello!

I just got WSL2 up and running on a new server - Windows 11 pro with 2 x 80 core Intel(R) Xeon(R) Platinum 8380 CPU @ 2.30GHz

In tracking the discussions - it seems there is a known issue of WSL (and associated docker containers) only able to use one socket at this time. However, i'm wondering if there is also a max number of CPUs that can be used?

I've set processors=80 in C:/Users/me/.wslconfig
And also set the hypervisor scheduler to classic.

However, after wsl shutdown and restart, Ubuntu lscpu only shows 64 CPUs ( 1 socket, 32 cores, 2 threads )

If i set the processor count to a number less than 64, this is reflected in lscpu, however, anything above 64 seems to cap at 64.

I note that
zcat /proc/config.gz | grep NR_CPUS_DEFAULT
yields:
CONFIG_NR_CPUS_DEFAULT=64

Is this a cap?

If so, given this is a kernel file, not sure how to bump this up to 80 without rebuilding the kernel.

Any insights welcome, would love to get all 80 cores on one of the sockets working in WSL2/docker.

Cheers!

Edit:

I'm finding it's not just 64 - but the number of CPUs also are sometimes set to 32. If i shutdown and restart wsl - and check lscpu on Ubuntu with processors=80 in C:/users/me/.wsconfig - sometimes i get 64 and sometimes i get 32. I have not been able to discern what is driving this - seems random. Have tried both WSL version 2.0.14 and 2.0.9

32 cores out of a total of 160 available is problematic. Is there something i should be doing differently and/or is there a WSL fix here? Or is my recourse VMware and/or dual-boot to Linux?

Update (5/7/24)

I ended up installing Ubuntu dual boot on the server - can now run all 160 cores via docker with no issues on the Ubuntu partition.

**Comment by fengbopu** [comment]
Hi,
I have the same problem like you, can you tell me some specific methods to solve it? How link two  dual-boot wsls?
thanks!

**Comment by fengbopu** [comment]
Hi,
I try to run mpi on WSL (both mpiexec -n 64 and 100), I find -n 100 is faster, I'm  puzzled. Don't  64  is the capping?
Cheers

**Comment by felgryn** [comment]
I ended up moving to a dual-boot solution of windows and Ubuntu. Have been running mostly in the Ubuntu partition and have had no issues using all core on both sockets with Ubuntu.
